smap_l2_gridder/grid.py

```diff
@@ -130,67 +138,79 @@ def default_fill_value(data_type: np.dtype | None) -> np.integer | np.floating |
+def locate_row_and_column_for_group(
+    in_dt: DataTree, group: str, short_name: str
+) -> dict[str, DataArray]:
+    """Return the row and column information for this group.
 
-    TODO [MHS, 11/06/2024] This might go in a different file later with logic to
-    suss which grid each node is representing.
+    Use the short_name to determine the correct location of the row and column
+    indices variables within in the input DataTree structure.
 
     """
-    src_grid_info = {}
-    row = in_data[f'/{node}/EASE_row_index']
-    column = in_data[f'/{node}/EASE_column_index']
-    src_grid_info['rows'] = row.astype(row.encoding.get('dtype', 'uint16'))
-    src_grid_info['cols'] = column.astype(column.encoding.get('dtype', 'uint16'))
-
-    grid_info = {}
-    grid_info['src'] = src_grid_info
-    grid_info['target'] = get_target_grid_information(node)
+    info = get_collection_group_info(short_name, group)
+    row = in_dt[info['row']]
+    column = in_dt[info['col']]
+    return {
+        'rows': row.astype(row.encoding.get('dtype', 'uint16')),
+        'cols': column.astype(column.encoding.get('dtype', 'uint16')),
+    }
 
-    return grid_info
 
+def get_target_grid_information(group: str, short_name: str) -> dict:
+    """Return the target grid informaton.
 
-def get_target_grid_information(node: str) -> dict:
-    """Return the target grid informaton."""
-    if is_polar_node(node):
-        gpd_name = 'EASE2_N09km.gpd'
-        wkt = EPSG_6931_WKT
-    else:
-        gpd_name = 'EASE2_M09km.gpd'
-        wkt = EPSG_6933_WKT
+    Using the group name and collection short name look up the gpd and epsg
+    code that represent the target grid.
 
-    target_grid_info = parse_gpd_file(gpd_name)
-    target_grid_info['wkt'] = wkt
+    """
+    info = get_collection_group_info(short_name, group)
+    target_grid_info = parse_gpd_file(info['gpd'])
+    target_grid_info['wkt'] = CRS(info['epsg']).to_wkt()
     return target_grid_info
 
 
-def is_polar_node(node: str) -> bool:
-    """If the node name ends with "_Polar" it's the Northern Hemisphere data."""
-    return node.endswith('_Polar')
+def get_data_groups(in_data: DataTree) -> set[str]:
+    """Returns list of input groups containing griddable data."""
+    return set(in_data['/'].children) - set(get_metadata_children(in_data))
 
 
 def get_metadata_children(in_data: DataTree) -> list[str]:
-    """Fetch nodes with metadata.
+    """Fetch list of groups containing only metadata.
 
-    List of top level datatree children containing metadata to transfer
-    directly to output file.
+    List of top level datatree children containing only metadata to transfer
+    directly to output file without any changes.
 
-    Note: This returns a constant for now, but may require reading the in_data
-    in the future.
     """
-    return ['Metadata']
+    collection_config = get_collection_info(get_collection_shortname(in_data))
+    return collection_config['metadata']
```

Comment:
This still just returns ['Metadata'], but it felt like it was fine to leave it in the collection configuration.